pkg/operator/observe_config.go

```diff
@@ -107,48 +136,59 @@ func (c ConfigObserver) sync() error {
 
 // observeEtcdEndpoints reads the etcd endpoints from the endpoints object and then manually pull out the hostnames to
 // get the etcd urls for our config. Setting them observed config causes the normal reconciliation loop to run
-func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, []error) {
 	etcdURLs := []string{}
 	etcdEndpoints, err := kubeClient.CoreV1().Endpoints(etcdNamespaceName).Get("etcd", metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		return observedConfig, nil
+		return observedConfig, []error{fmt.Errorf("endpoints/etcd.kube-system: not found")}
 	}
 	if err != nil {
-		return observedConfig, err
+		return observedConfig, []error{err}
+	}
+	dnsSuffix, ok := etcdEndpoints.Annotations["alpha.installer.openshift.io/dns-suffix"]
+	if !ok || dnsSuffix == "" {
+		return observedConfig, []error{fmt.Errorf("endpoints/etcd.kube-system: alpha.installer.openshift.io/dns-suffix annotation not found")}
 	}
-	for _, subset := range etcdEndpoints.Subsets {
-		for _, address := range subset.Addresses {
-			etcdURLs = append(etcdURLs, "https://"+address.Hostname+"."+etcdEndpoints.Annotations["alpha.installer.openshift.io/dns-suffix"]+":2379")
+	var errs []error
+	for subsetIndex, subset := range etcdEndpoints.Subsets {
+		for addressIndex, address := range subset.Addresses {
+			if address.Hostname == "" {
+				errs = append(errs, fmt.Errorf("endpoints/etcd.kube-system: subsets[%v]addresses[%v].hostname not found", subsetIndex, addressIndex))
+				continue
+			}
+			etcdURLs = append(etcdURLs, "https://"+address.Hostname+"."+dnsSuffix+":2379")
 		}
 	}
+	if len(errs) > 0 {
+		return observedConfig, errs
```

Comment:
I'm still missing my "use the old config", aren't I?

Reply:
> I'm still missing my "use the old config", aren't I?

This isn't strictly worse than what I had before, so I can take it as a followup.  A quick followup.